help_function/src/util_group_load_store_test.cpp

```diff
@@ -0,0 +1,199 @@
+  // dpct::group::store_algorithm::BLOCK_STORE_STRIPED in its entirety as API
+  // functions
+  sycl::queue q(dpct::get_default_queue());
+  oneapi::dpl::counting_iterator<int> count_it(0);
+  sycl::buffer<int, 1> buffer(count_it, count_it + 512);
+  
+  q.submit([&](sycl::handler &h) {
+    using group_load =
+        dpct::group::workgroup_load<4, T, int, const int *, sycl::nd_item<3>>;
+    using group_store =
+        dpct::group::workgroup_store<4, S, int, int *, sycl::nd_item<3>>;
+    size_t temp_storage_size = group_load::get_local_memory_size(128);
+    sycl::local_accessor<uint8_t, 1> tacc(sycl::range<1>(temp_storage_size), h);
+    sycl::accessor data_accessor_read_write(buffer, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          auto *d_r_w =
+              data_accessor_read_write.get_multi_ptr<sycl::access::decorated::yes>()
+                  .get();
+          auto *tmp = tacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          // Load thread_data of each work item to blocked arrangement
+          group_load(tmp).load(item, d_r_w, thread_data);
+          // Store thread_data of each work item from blocked arrangement
+          group_store(tmp).store(item, d_r_w, thread_data);
+          });
+  });
+  q.wait_and_throw();
+
+  sycl::host_accessor data_accessor(buffer, sycl::read_write);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  return helper_validation_function<T>(ptr, "test_group_load_store");
+}
+
+bool test_load_store_subgroup_striped_standalone() {
+  // Tests dpct::group::load_subgroup_striped as standalone method
+  sycl::queue q(dpct::get_default_queue());
+  int data[512];
+  for (int i = 0; i < 512; i++)
+    data[i] = i;
+  sycl::buffer<int, 1> buffer(data, 512);
+  sycl::buffer<uint32_t, 1> sg_sz_buf{sycl::range<1>(1)};
+  
+  q.submit([&](sycl::handler &h) {
+    sycl::accessor dacc_read_write(buffer, h, sycl::read_write);
+    sycl::accessor sg_sz_dacc(sg_sz_buf, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          auto *d_r_w =
+              dacc_read_write.get_multi_ptr<sycl::access::decorated::yes>().get();
+          auto *sg_sz_acc =
+              sg_sz_dacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          size_t gid = item.get_global_linear_id();
+          if (gid == 0) {
+            sg_sz_acc[0] = item.get_sub_group().get_local_linear_range();
+          }
+          dpct::group::uninitialized_load_subgroup_striped<4, int>(item, d_r_w,
+                                                                   thread_data);
+          dpct::group::store_subgroup_striped<4, int>(item, d_r_w, thread_data);

```

Comment:
Since these are loaded into uninitialized memory with a `new` command, I believe we also need to `delete` the memory.

Reply:
Added a thread destructor in this case, however for complex types - structs etc, would it be better to have destructor inside the main subgroup striped load class?